src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs

```diff
@@ -377,6 +377,56 @@ private static string GetAccessibilityModifiersRequiredEditorConfigString(CodeSt
                 KeyValuePairUtil.Create("all", UnusedParametersPreference.AllMethods),
             });
 
+        #region dotnet_style_prefer_foreach_explicit_cast_in_Source
+
+        private static readonly CodeStyleOption2<ForEachExplicitCastInSourcePreference> s_forEachExplicitCastInSourceNonLegacyPreference =
+            new(ForEachExplicitCastInSourcePreference.NonLegacy, NotificationOption2.Suggestion);
+
+        private static readonly BidirectionalMap<string, ForEachExplicitCastInSourcePreference> s_forEachExplicitCastInSourcePreferencePreferenceMap =
+            new(new[]
+            {
+                KeyValuePairUtil.Create("always", ForEachExplicitCastInSourcePreference.Always),
+                KeyValuePairUtil.Create("never", ForEachExplicitCastInSourcePreference.Never),
+                KeyValuePairUtil.Create("non_legacy", ForEachExplicitCastInSourcePreference.NonLegacy),
+            });
+
+        internal static readonly PerLanguageOption2<CodeStyleOption2<ForEachExplicitCastInSourcePreference>> ForEachExplicitCastInSource =
```

Comment:
`AutomationObject`  may need an update.

Reply:
enh... but does it really?

Reply:
@CyrusNajmabadi It affects importing and exporting VS settings. I recall @jmarolf added some missing options few years ago, so he might have a better idea. Usually I find myself adding options there.

Reply:
https://github.com/dotnet/roslyn/pull/16512 That was the PR.

Reply:
I added an option the other day. I added code to `AutomationObject` in the same PR. I missed seeing you comment on my PR @Youssef1313 😁

Reply:
Yes, if the expectation is that this option is round-trip-able in Visual Studio then support for it needs to be in the AutomationObject since that is what allows us to serialize VS Settings xml

Reply:
i wasn't intending to add a VS option for this.  This seemed like a very niche thing to have a user visible option on.  I think this is more for users who run into this issue and decide to opt in at the editorconfig level.  But we can def discuss this on monday if people feel differently :)

Reply:
@CyrusNajmabadi In that case, I wonder why there is a `RoamingProfileStorageLocation` for this option?

Reply:
yup.  that makes sense.  will fixup.  thanks! :)